drivers/bridge/port_mapping.go

```diff
@@ -25,11 +27,23 @@ func (n *bridgeNetwork) allocatePorts(ep *bridgeEndpoint, reqDefBindIP net.IP, u
 		defHostIP = reqDefBindIP
 	}
 
-	return n.allocatePortsInternal(ep.extConnConfig.PortBindings, ep.addr.IP, defHostIP, ulPxyEnabled)
+	var pb []types.PortBinding
+
+	if ep.addrv6 != nil {
```

Comment:
if a network will be either a ipv4 or a ipv6 network, why do we need to allocate both kind of ports here

Reply:
I am not sure if it is possible to have a network without an IPv4 address.
But otherwise a network can be a only IPv4 network or a IPv6/IPv4 (dual stack) network.

In case of a dual stack network the port is allocated on both versions.
For example in case of an web server container that exposes port 80 it is allocated as `0.0.0.0:80` and `[::]:80`.

If IPv6 only networks are also possible this should be changed of course.

Reply:
can we take care of this case later, and only pick one binding for now
if ep.addrv6 != nil {
// bind to ipv6 addr
} else {
// bind to ipv4 addr
}

Reply:
If it would be implemented this way it would cause only a forward for IPv6 which break the IPv4 behavior.

I have reworked this section a bit and also handle the `EnableIP6Tables` correctly.
I also changed the signature back and handle the different calls completely outside.
Hope this is a proper way to handle this.